lms/djangoapps/shoppingcart/models.py

```diff
@@ -1606,3 +1609,355 @@ def analytics_data(self):
             data['name'] = settings.PLATFORM_NAME
             data['category'] = settings.PLATFORM_NAME
         return data
+
+
+TRANSACTION_TYPE_PURCHASE = 'p'
+TRANSACTION_TYPE_REFUND = 'r'
+
+
+class PaymentTransaction(TimeStampedModel):
+    """
+    This model will act as a copy of all transaction information that is stored on any Payment Processor. We
+    have a syncronization management command to extract all of the daily reports and store
+    locally in our database
+    """
+
+    # The primary identifier for any transaction as stored in
+    # the payment processor.
+    remote_transaction_id = models.CharField(max_length=64, db_index=True, unique=True)
+
+    # any 'account identified' (aka Merchant Account) associated with this transaction
+    # note, this is an optional field
+    account_id = models.CharField(max_length=32, null=True)
+
+    # The timestamp associated with when the payment processing
+    # occured
+    processed_at = models.DateTimeField(db_index=True)
+
+    # the Order model that this processing is associated with
+    order = models.ForeignKey(Order, db_index=True)
+
+    currency = models.CharField(max_length=16)
+    amount = models.DecimalField(decimal_places=2, max_digits=30)
+
+    # transaction_type in Payment Processing report is
+    # whether it is a purchase or a refund: 'purchase' or 'refund'
+    TRANSACTION_TYPE_CHOICES = (
```

Comment:
Will there be any other types of transactions? If it is just Purchase or Refund, we could represent this with positive and negative values in the transaction amount.  The type could be determined by evaluating the data, avoiding the risk of a stateful field coming out of sync with the rest of the data model.